pkg/didcomm/protocol/didexchange/states.go

```diff
@@ -528,6 +528,33 @@ func (ctx *context) didDocAttachment(doc *did.Doc, myVerKey string) (*decorator.
 	return docAttach, nil
 }
 
+func (ctx *context) resolvePublicKey(kid string) ([]byte, error) {
+	if strings.HasPrefix(kid, "did:key:") {
```

Comment:
Is this logic specific to Interop as this is called inside if `ctx.doACAPyInterop` is true ?

Reply:
It's not technically interop-only, but it's only needed in a specific case during aca-py interop.

Lifting it out requires some refactoring to change how we handle our key references across more of didexchange, and that wasn't necessary here so I avoided it.